dipy/direction/tests/test_bootstrap_direction_getter.py

```diff
@@ -0,0 +1,271 @@
+    toydict = {}
+
+    hsph_updated = HemiSphere.from_sphere(unit_icosahedron).subdivide(2)
+
+    vertices = hsph_updated.vertices
+
+    bvecs = vertices
+    bvals = np.ones(len(vertices)) * 1000
+
+    bvecs = np.insert(bvecs, 0, np.array([0, 0, 0]), axis=0)
+    bvals = np.insert(bvals, 0, 0)
+
+    gtab = gradient_table(bvals, bvecs)
+
+    toy_voxel = single_tensor(gtab)
+    toy_data = np.tile(toy_voxel, (11, 11, 11, 1))
+    toy_tissue_classifier = np.ones(toy_data.shape[:-1])
+    toy_affine = np.eye(4)
+
+    # make a slice roi that should track in-plane only
+    toy_roi_long_plane = np.zeros(toy_data.shape[:-1])
+    toy_roi_long_plane[:, :, 0]=1
+    # make a slice roi that should track directly across the whole volume
+    toy_roi_radial_plane = np.zeros(toy_data.shape[:-1])
+    toy_roi_radial_plane[0, :, :] = 1
+    # make an roi that contains the center voxel only
+    toy_roi_center_vox = np.zeros(toy_data.shape[:-1])
+    toy_roi_center_vox[5, 5, 5] = 1
+
+    toydict['gtab'] = gtab
+    toydict['toy_data'] = toy_data
+    toydict['toy_affine'] = toy_affine
+    toydict['toy_roi_long_plane'] = toy_roi_long_plane
+    toydict['toy_roi_radial_plane'] = toy_roi_radial_plane
+    toydict['toy_roi_center_vox'] = toy_roi_center_vox
+    toydict['toy_tissue_classifier'] = toy_tissue_classifier
+
+    return toydict
+
+
+def test_bdg_initial_direction():
+    # test that we get one direction when we have a single tensor
+    toydict = {}
+
+    hsph_updated = HemiSphere.from_sphere(unit_icosahedron).subdivide(2)
+
+    vertices = hsph_updated.vertices
+
+    bvecs = vertices
+    bvals = np.ones(len(vertices)) * 1000
+
+    bvecs = np.insert(bvecs, 0, np.array([0, 0, 0]), axis=0)
+    bvals = np.insert(bvals, 0, 0)
+
+    gtab = gradient_table(bvals, bvecs)
+
+    toy_voxel = single_tensor(gtab).reshape([1, 1, 1, -1])
+
+    model = dti.TensorModel(gtab)
+
+    direction_getter = bdg.BootDirectionGetter.from_data(toy_voxel, model, 30)
+
+    initial_direction = direction_getter.initial_direction(np.zeros(3))
+
+    npt.assert_equal(len(initial_direction), 1)
+
+    npt.assert_allclose(initial_direction[0], [1, 0, 0], atol=0.1)
+
+    # test that we get multiple directions when we have a multi-tensor
+
+    mevals = np.array([[1.5, 0.4, 0.4], [1.5, 0.4, 0.4]]) * 1e-3
+    fracs = [60, 40]
+    toy_voxel, primary_evecs = multi_tensor(gtab, mevals,
+                                            fractions=fracs, snr=None)
+    toy_voxel = toy_voxel.reshape([1, 1, 1, -1])
+    model = csd.ConstrainedSphericalDeconvModel(gtab, response=None)
+    direction_getter = bdg.BootDirectionGetter.from_data(toy_voxel, model, 30)
+    initial_direction = direction_getter.initial_direction(np.zeros(3))
+    print(initial_direction)
+    npt.assert_equal(len(initial_direction), 2)
+    npt.assert_allclose(initial_direction, primary_evecs, atol=0.1)
+
+
+def test_bdg_get_direction():
+    ''' test if direction getter goes 5 tries if '''
```

Comment:
triple double quoted strings should be used for docstrings

Reply:
fixed